src/Singularity/ContainerBuilder.cs

```diff
@@ -211,6 +211,23 @@ public void Register(Type serviceType, Type[] implementationTypes, Action<Weakly
             }
         }
 
+        /// <summary>
+        /// Registers services defined in an <see cref="IModule"/>.
+        /// </summary>
+        /// <param name="module">The <see cref="IModule"/> to register services from</param>
+        /// <exception cref="ArgumentNullException">If <paramref name="module"/> is <c>null</c></exception>
+        /// <returns></returns>
+        public void RegisterModule(IModule module)
+        {
+            if (module == null) throw new ArgumentNullException(nameof(module));
+            Registrations.CurrentModule = module;
```

Comment:
Not your fault since I made this property like this but having these magic properties that have to be set and unset like so might not have been the best design decision. I think its fine to leave it like this for now though.